cpp/tests/community/k_truss_test.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include "utilities/base_fixture.hpp"
+#include "utilities/check_utilities.hpp"
+#include "utilities/conversion_utilities.hpp"
+#include "utilities/test_graphs.hpp"
+#include "utilities/thrust_wrapper.hpp"
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/graph.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/utilities/high_res_timer.hpp>
+
+#include <raft/core/handle.hpp>
+#include <raft/util/cudart_utils.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+#include <numeric>
+#include <random>
+#include <set>
+#include <vector>
+
+struct KTruss_Usecase {
+  int32_t k_{3};
+  bool test_weighted_{false};
+  bool check_correctness_{true};
+};
+
+template <typename input_usecase_t>
+class Tests_KTruss : public ::testing::TestWithParam<std::tuple<KTruss_Usecase, input_usecase_t>> {
+ public:
+  Tests_KTruss() {}
+
+  static void SetUpTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  // FIXME: There is an utility equivalent functor not
+  // supporting host vectors.
+  template <typename type_t>
+  struct host_nearly_equal {
+    const type_t threshold_ratio;
+    const type_t threshold_magnitude;
+
+    bool operator()(type_t lhs, type_t rhs) const
+    {
+      return std::abs(lhs - rhs) <
+             std::max(std::max(lhs, rhs) * threshold_ratio, threshold_magnitude);
+    }
+  };
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  std::tuple<std::vector<vertex_t>, std::vector<vertex_t>, std::optional<std::vector<weight_t>>>
+  k_truss_reference(std::vector<vertex_t> h_offsets,
+                    std::vector<vertex_t> h_indices,
+                    std::optional<std::vector<weight_t>> h_values,
```

Comment:
No need to address this in this PR, but just fyi,

You're passing h_offsets, h_indices, and h_values by value and based on the size of the vectors, this can be expensive. You can pass a const reference (`const &`) instead.

Reply:
Right. I will resolve it in my next PR